module/Photo/src/Photo/Service/Photo.php

```diff
@@ -220,7 +221,74 @@ public function movePhoto($photoId, $albumId)
         return true;
 
     }
-
+    /**
+     * Generates the PhotoOfTheWeek and adds it to the list
+     * if at least one photo has been viewed in the specified time. 
+     * The parameters determine the week to check the photos of.  
+     * 
+     * @param \DateTime $begindate
+     * @param \DateTime $enddate
+     * 
+     * @return \Photo\Model\Photo|null
+     */
+    public function generatePhotoOfTheWeek($begindate, $enddate)
+    {
+        $bestPhoto = $this->determinePhotoOfTheWeek($begindate, $enddate);
+        if (is_null($bestPhoto)){
+            return null;
+        }
+        $weeklyPhoto = new WeeklyPhotoModel();
+        $weeklyPhoto->setPhoto($bestPhoto);
+        $weeklyPhoto->setWeek($begindate);
+        $mapper = $this->getWeeklyPhotoMapper();
+        $mapper->persist($weeklyPhoto);
+        $mapper->flush();
+        return $bestPhoto;
+    }
+    
+    /**
+     * Determine which photo is the photo of the week
+     * 
+     * @param \DateTime $begindate
+     * @param \DateTime $enddate
+     * @return \Photo\Model\Photo|null
+     */
+    private function determinePhotoOfTheWeek($begindate, $enddate)
+    {
+        $results = $this->getHitMapper()->getHitsInRange($begindate, $enddate);
+        if (empty($results)){
+            return null;
+        }
+        $bestPref = -1;
+        foreach ($results as $res){
+            $photo = $this->getPhotoMapper()->getPhotoById($res[1]);
+            if (!$this->getWeeklyPhotoMapper()->hasBeenPhotoOfTheWeek($photo)
+                && $this->photoPreference($photo, $res[2])
+                    > $bestPref){
+                $bestPhoto = $photo;
+                $bestPref = $this->photoPreference($photo, $res[2]);
+            }
+        }
+        return $bestPhoto;
+    }
+    
+    
+    /**
+     * Determine the preference rating of the photo.
+     * 
+     * @param integer $photoId
+     * @param integer $occurences
+     * @return float
+     */
+    private function photoPreference($photo, $occurences)
```

Comment:
Why is this private? Actually, why are so many methods here private? Unless explicitly needed, the PSR standards state that functions should be protected.